snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/tracker/Tracker.java

```diff
@@ -957,4 +964,27 @@ public void removeGlobalContext(String tag) {
             }
         }
     }
+
+    // gdpr context
+    private boolean gdprContext = false;
+    private String basisForProcessing;
+    private String documentId;
+    private String documentVersion;
+    private String documentDescription;
+
+    /**
+     * Enables gdpr context to be sent with every event
+     */
+    public void enableGDPRContext(BASIS basisForProcessing, String documentId, String documentVersion, String documentDescription) {
+        String basis = basisForProcessing.toString().toLowerCase();
```

Comment:
Since this is an [enum](https://github.com/snowplow/iglu-central/blob/248b3be05b097693363e014b902008fa6cb3bfc8/schemas/com.snowplowanalytics.snowplow/gdpr/jsonschema/1-0-0#L14) in the schema, we've opted to [make it an enum in the javascript tracker](https://github.com/snowplow/snowplow-javascript-tracker/blob/a2a12a5c1a2065db8df250eb1bf6e31900e81a0c/src/js/tracker.js#L111).

Not sure how practicable that is for Android